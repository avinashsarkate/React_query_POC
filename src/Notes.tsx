import React from "react";
import { useQuery, useMutation, useQueryClient, InvalidateQueryFilters } from "@tanstack/react-query";

import { NoteStyled } from "./Notes.styled";
import { deleteNotes, fetchNotes } from "./API";

interface Note {
  id: string;
  title: string;
  body: string;
}

 export const Notes: React.FC = () => {
  const queryClient = useQueryClient();
  const { isLoading, error, data } = useQuery<Note[]>({
    queryKey: ['notes'],
    queryFn: fetchNotes
  });
  
  const deleteNoteMutation = useMutation<void, unknown, string>(
    {
      mutationFn:(id)=> deleteNotes(id),
      onSuccess: () => {
        queryClient.invalidateQueries("notes" as InvalidateQueryFilters);
      },
    }
  );
  

  const handleDelete = (id: string) => {
    deleteNoteMutation.mutate(id);
  };

  if (isLoading) return <div>Loading...</div>;
  if (error) return <div>Error: {error.message}</div>;

  return (
    <NoteStyled>
      <div>
        <h1>Notes List</h1>
        <div className="note-container">
          {data &&
            data.map((note) => (
              <div className="note-box" key={note.id}>
                <h3>{note.title}</h3>
                <p>{note.body}</p>
                <button onClick={() => handleDelete(note.id)}>Delete</button>
              </div>
            ))}
        </div>
      </div>
    </NoteStyled>
  );
};


