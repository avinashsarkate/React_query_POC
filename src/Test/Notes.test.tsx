

/* eslint-disable testing-library/prefer-screen-queries */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { Notes } from "../Notes";


jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: {} })),
 delete:jest.fn(() => Promise.resolve({ data: {} })),
 post: jest.fn(() => Promise.resolve({ data: {} }))
}))


describe("Notes Component", () => {

  beforeEach(() => {
    jest.clearAllMocks();
  })


  const queryClient = new QueryClient();

  const component = (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Notes />
      </BrowserRouter>
    </QueryClientProvider>
  );


  test('renders loading state', () => {
    jest.requireMock('@tanstack/react-query').useQuery.mockReturnValueOnce({ isLoading: true });
    render(component);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  })


  test("Render component without crashing", () => {
    render(component);
  });
 ;
  test("Render component", () => {
    const { container } = render(component);
    expect(container).toBeTruthy();
  });

  test("to create snapshot", () => {
    const { asFragment } = render(component);
    expect(asFragment).toMatchSnapshot();
  });
});


