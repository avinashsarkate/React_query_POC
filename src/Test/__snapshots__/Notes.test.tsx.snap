// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Component to create snapshot 1`] = `[Function]`;
